core/src/main/scala/org/apache/spark/ui/WebUI.scala

```diff
@@ -91,7 +91,12 @@ private[spark] abstract class WebUI(
   /** Attach a handler to this UI. */
   def attachHandler(handler: ServletContextHandler) {
     handlers += handler
-    serverInfo.foreach(_.addHandler(handler))
+    serverInfo.foreach { sInfo =>
+      sInfo.addHandler(handler)
+      // If the UI has already been bound, we need to add the filters to the newly attached
+      // handlers. Otherwise, they will be attached when binding.
+      addFilters(Seq(handler), conf)
```

Comment:
There's a call to `addFilters` in `HistoryServer.scala` that becomes redundant with this code, I think.

Reply:
Nevermind you already removed it.